light-prover/prover/batch_append_circuit_test.go

```diff
@@ -0,0 +1,309 @@
+package prover
+
+import (
+	"math/big"
+	"testing"
+
+	"github.com/consensys/gnark-crypto/ecc"
+	"github.com/consensys/gnark/frontend"
+	"github.com/consensys/gnark/test"
+)
+
+func TestBatchAppendCircuit(t *testing.T) {
+	assert := test.NewAssert(t)
+
+	t.Run("Successful proofs", func(t *testing.T) {
+		testCases := []struct {
+			name       string
+			treeDepth  uint32
+			batchSize  uint32
+			startIndex uint32
+		}{
+			{"Small batch", 26, 10, 0},
+			{"Medium batch", 26, 100, 0},
+			{"Large batch", 26, 1000, 0},
+			{"Tree depth = 4", 4, 10, 0},
+			{"Non-zero start index", 26, 100, 500},
+			{"Start index near tree end", 10, 10, (1 << 10) - 15}, // 2^10 - 15 = 1009
```

Comment:
Imo this should not be possible. When integrated with a smart contract which provides this input it is not possible but I am wondering whether we can construct the circuit so that it is also not possible there.
We could for example add next index to the subtree hashchain.